System/Process.hs

```diff
@@ -245,7 +246,10 @@ withCreateProcess_ fun c action =
     C.bracketOnError (createProcess_ fun c) cleanupProcess
                      (\(m_in, m_out, m_err, ph) -> action m_in m_out m_err ph)
 
-
+-- | Cleans up the process.
+-- 
+-- This function is meant to be invoked from any application level cleanup 
+-- handler. It terminates the process, and closes any 'CreatePipe' 'handle's.
```

Comment:
LGTM. Can you add a `@since` comment with a new minor version and update the `ChangeLog` with an entry about this new function?

Reply:
@snoyberg Done.